native/cocos/renderer/gfx-metal/MTLPipelineState.mm

```diff
@@ -304,10 +304,11 @@ of this software and associated engine source code (the "Software"), a limited,
             depthStencilTexIndex = subpass.depthStencil;
             for (size_t i = 0; i < subpass.inputs.size(); ++i) {
                 uint32_t input = subpass.inputs[i];
+
                 if (inputs.find(input) == inputs.end()) {
                     inputs.insert(input);
-                    if(input >= colorAttachments.size()) {
-                        depthStencilTexIndex = input;
+                    if(_renderPass->getColorAttachments()[input].format == Format::DEPTH ||
+                       _renderPass->getColorAttachments()[input].format == Format::DEPTH_STENCIL) {
```

Comment:
Format::STENCIL as well ?

Reply:
adding.

Reply:
Single stencil will update in another pr later.